kong/runloop/handler.lua

```diff
@@ -360,25 +360,53 @@ local function register_events()
 
     -- declarative config updates
 
-    worker_events.register(function(default_ws)
+    local current_router_hash
+    local current_plugins_hash
+    local current_balancer_hash
+
+    worker_events.register(function(data)
       if ngx.worker.exiting() then
         log(NOTICE, "declarative flip config canceled: process exiting")
         return true
       end
 
+      local default_ws
+      local router_hash
+      local plugins_hash
+      local balancer_hash
+
+      if type(data) == "table" then
+        default_ws = data[1]
+        router_hash = data[2]
+        plugins_hash = data[3]
+        balancer_hash = data[4]
+      end
+
       local ok, err = concurrency.with_coroutine_mutex(FLIP_CONFIG_OPTS, function()
-        balancer.stop_healthcheckers(CLEAR_HEALTH_STATUS_DELAY)
+        if balancer_hash == nil or balancer_hash ~= current_balancer_hash then
```

Comment:
as `if balancer_hash == nil or balancer_hash ~= current_balancer_hash` happens twice not constinously, to avoid us introduce bugs of modifying the hash variable in between, i would suggest we to memorize the result first, and use the memorized result in both L386 and L406, so those two branches are guranteed to be both executed or non-excuted.
```suggestion
       local should_rebuild_balancer = balancer_hash == nil or balancer_hash ~= current_balancer_hash
        if should_rebuild_balancer then
```